src/boat/private/Config.nim

```diff
@@ -1,10 +1,91 @@
 ##
 ## Config
 ## ======
-## base configuration for Manifest and CaptainsLog
+## provides a parser and standard interface
 
 ##[
 ## TLDR
-- a manifest eventually becomes the captains log
-  - from another perspective, the captains log is the realization of a manifest
+- come back later
+
+links
+-----
+- [parsecfg](https://nim-lang.org/docs/parsecfg.html)
+
+Interface
+=========
+- come back later
 ]##
+
+import std/[
+  os,
+  parsecfg,
+  strutils,
+]
+
+type Config* = ref object of RootObj
+  cachedir*: string ## \
+    ## defaults to getCacheDir()
+  use*: string ## \
+    ## path on disk pointing to a manifest / or dir with a manifest
+    ## URL pointing to a *.ini file
+  #--- private ---#
+  parsed: parsecfg.Config ## \
+    ## the parsed config after loading
+  saved: bool ## \
+    ## true if self.use has been saved to disk
+
+proc parseLocalManifest*(self: Config, path = self.use): bool =
+  ## parse self.use to self.parsed
+  ## prefer calling self.load or self.reload for validation
+  self.parsed = loadConfig path
+  result = true
+
+proc save*(self: Config): bool =
+  ## serialize Self.parsed to disk @ self.cachedir | getCachDir() / <SELF.ID>.manifest.nim.ini
+  result = true
+
+proc reload*(self: Config): bool =
+  # starts with https?
+    # ends in .ini? break
+    # throw: urls must point to a manifest.nim.ini
+  # must be a filepath, as we only support https
+    # do we have read access?
+      # does it end in .ini ? break
+      # must be a dir
+        # does it contain a manifest.nim.ini ? break
+    # throw: couldnt find / or read a *.ini file
+  # manifest seems to be okay! lets do the actual loading
+    # parse and upsert to self.parsed
+      # if manifest.nim.ini loads other manifests, recurse
+    # save parsed to disk
+  # everything must be okay!
+  case self.use.startsWith "https"
+    of true: raise newException(CatchableError, "TODO: remote manifests not setup")
+    else:
+      try:
+        # this throws if its not a file; no need to check for http/ftp/etc
+        let path = self.use.getFileInfo
+        case path.kind
+          of pcFile, pcLinkToFile:
+            if fpUserRead notin path.permissions:
+              raise newException(CatchableError, "invalid file permissions")
+            elif not self.use.endsWith ".ini":
+              raise newException(CatchableError, "invalid file type")
+            elif not self.parseLocalManifest():
+              raise newException(CatchableError, "cant parse config")
+          of pcDir, pcLinkToDir:
+            raise newException(CatchableError, "TODO: loading from dir not setup")
+      except CatchableError:
+        debugEcho repr getCurrentException()
+        raise newException(CatchableError, "unable to load conf from disk")
+  # TODO: pretty sure ADRs require saving to captains.log as well
```

Comment:
we didnt spend enough time thinking through the caching strategy 
it should look like
- captainslog
  - contains > actions taken that impact runtime logic
  - points to > cachedDir/cached files
  - etc

the idea is not to bloat captains log as it will likely be write heavy and read heavy at runtime
- anything thats not critical to runtime usage should be in a separate cached file
  - e.g. all the intialization logic in Config

Reply:
update ADRs to include a breakdown of Boat's lifecycle
that will help cement what needs to go where